R/calc_clutchtype.R

```diff
@@ -23,393 +25,798 @@
 #'                            sample(seq(1, 31, 1), size = 100, replace = TRUE), sep = "-"),
 #'                            format = "%Y-%m-%d"))
 #' calc_clutchtype(data = dat, na.rm = FALSE)
-calc_clutchtype <- function(data, na.rm = FALSE) {
+calc_clutchtype <- function(data,
+                            protocol_version = "1.0",
+                            na.rm = FALSE) {
 
-  if ("LayDate_observed" %in% colnames(data)) {
+  if (protocol_version == "1.0"){
 
-    cutoff_dat <- data %>%
-      dplyr::group_by(PopID, BreedingSeason, Species) %>%
-      dplyr::mutate(cutoff = tryCatch(expr = min(.data$LayDate_observed, na.rm = TRUE) + lubridate::days(30),
-                                      warning = function(...) return(NA))) %>%
-      # Determine brood type for each nest based on female ID
-      dplyr::group_by(BreedingSeason, Species, FemaleID)
+    if ("LayDate_observed" %in% colnames(data)) {
```

Comment:
This is true. The function was actually already written to handle both the new and old protocol versions, but did so by looking for 'LayDate_ovserved' in the column names and processing the data accordingly. I've changed the function so that it now uses the column names based on the version argument. This required changing column names in the pipelines that were formatted to version 1.1 since all of these pipelines first put the column names in the old format and then renamed the columns to the new format before exporting...